core/src/com/po/fuck/model/Player.java

```diff
@@ -26,6 +28,9 @@ public class Player extends Entity {
                         new BasicMovement(this, DEFAULT_SPEED),
                         Constants.BOOST_DISTANCE, Constants.BOOST_DURATION, Constants.BOOST_COOLDOWN));
         teamTag = Constants.PLAYER_TEAM_TAG;
+        BasicSpriteInfo info = SpriteInfoManager.getBasicSpriteInfo(this.getClass());
+        geometryData.setHeight(info.getFrameHeight());
+        geometryData.setWidth(info.getFrameWidth());
```

Comment:
Also why do constructor still accepts the full `GeometryData`, and simply rewriting some fields ? It should accept `Vector2 position` or maybe some `PositionData {Vec2 position; float rotation; };` and make `GeometryData extends PositionData`

Reply:
Maybe make constructor private, and add a `static create(...) -> Player`.

Reply:
I will change constructor to ```Player(Vector2 position, float rotation)``` and do ```super (new GeometryData(..., BasicSpriteManager.getBasicSpriteInfo(Player.class).getSize(), ...), ...)```. Is it will be OK?